core/shared/src/main/scala/zio/FiberRefs.scala

```diff
@@ -26,50 +27,78 @@ import scala.annotation.tailrec
  * example between an asynchronous producer and consumer.
  */
 final class FiberRefs private (
-  private[zio] val fiberRefLocals: Map[FiberRef[_], (::[(FiberId.Runtime, Any, Int)], Int)]
+  private[zio] val fiberRefLocals: Map[FiberRef[_], FiberRefs.Value]
 ) { self =>
+  import zio.FiberRefs.{StackEntry, Value}
 
   /**
    * Returns a new fiber refs with the specified ref deleted from it.
    */
-  def delete(fiberRef: FiberRef[_]): FiberRefs =
-    FiberRefs(fiberRefLocals - fiberRef)
+  def delete(fiberRef: FiberRef[_]): FiberRefs = {
+    val newMap = fiberRefLocals - fiberRef
+    if (newMap eq fiberRefLocals) self
```

Comment:
There was interesting comments made on Twitter by [@nilskp](https://twitter.com/nilskp):

See: 
- https://x.com/nilskp/status/1779946801841275224
- https://x.com/nilskp/status/1779969123113439442

I'm not sure whether this is documented or not, but we rely heavily on this kind of behaviour in your work. 
It seems to me that if Scala changes its behaviour (on purpose or by mistake), it'll potentially introduce hard-to-understand bugs. 

What do you think about writing tiny tests in your PRs insuring that all the operators your using before to make references comparison are working as expected?
Example of such test (in pseudo-code):
```scala
test("m0: Map(1 -> 2) - m1: Map(2 -> 3) eq m0") {
  val m0 = Map(1 -> 2)
  val m1 = Map(2 -> 3)
  
  assertTrue((m0 - m1) eq m0)
}
```

Same with `transform` that you use somewhere else and all the potential operators that you're using before to tests collection references
WDYT?

Reply:
In this case, it won't be a bug, just a performance regression. I'm fine with this shortcut, assuming `-` actually does the sensible thing and returns the same map if nothing is removed (one cannot take anything for granted with Scala collections; see also performance of array extensions 😄 ).